app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -158,6 +160,87 @@ public StudyParticipant getParticipant(Study study, String email) {
         }
         return accountDao.getPagedAccountSummaries(study, offsetBy, pageSize, emailFilter);
     }
+
+    /**
+     * Create a study participant. A password must be provided, even if it is added on behalf of a 
+     * user before triggering a reset password request.  
+     */
+    public void createParticipant(Study study, StudyParticipant participant) {
+        checkNotNull(study);
+        checkNotNull(participant);
+        
+        Validate.entityThrowingException(new StudyParticipantValidator(study, true), participant);
+
+        externalIdService.reserveExternalId(study, participant.getExternalId());
+        
+        SignUp signUp = new SignUp(participant.getEmail(), participant.getPassword(), null, null);
+        Account account = accountDao.signUp(study, signUp, study.isEmailVerificationEnabled());
+        
+        // This should not fail as we're now creating the mapping in sign-up itself. User's email
+        // doesn't even need to be verified before we can add this information to the account.
+        HealthId healthId = healthCodeService.getMapping(account.getHealthId());
+        String healthCode = healthId.getCode();
+        
+        Map<ParticipantOption,String> options = Maps.newHashMap();
+        for (ParticipantOption option : ParticipantOption.values()) {
+            options.put(option, option.fromParticipant(participant));
+        }
+        if (study.isExternalIdValidationEnabled()) {
+            options.remove(EXTERNAL_IDENTIFIER);
+        }
+        optionsService.setAllOptions(study.getStudyIdentifier(), healthCode, options);
+        
+        account.setFirstName(participant.getFirstName());
+        account.setLastName(participant.getLastName());
+        for(String attribute : study.getUserProfileAttributes()) {
+            String value = participant.getAttributes().get(attribute);
+            account.setAttribute(attribute, value);
+        }
+        accountDao.updateAccount(study, account);
+        
+        externalIdService.assignExternalId(study, participant.getExternalId(), healthCode);
+    }
+    
+    public void updateParticipant(Study study, String email, StudyParticipant participant) {
```

Comment:
createParticipant and updateParticipant are pretty similar, can we refactor?

Are there reasons for subtle differences such as
- calling assignExternalId in a different order
- updateParticipant checks isExternalIdValidationEnabled, createParticipant does not
- updateParticipant requires email and healthCode

Seems like there's an intended different use case that I'm missing (not just create vs update, but additional rules/validation)